functions/index.ts

```diff
@@ -1,168 +1,105 @@
     default:
-      HttpResponse({statusCode: 400, body: JSON.stringify("Unable to process request")});
+      callback(null, HttpResponse({statusCode: 400, body: JSON.stringify("Invalid operation.")}));
       break;
-  }
-  
-  async function getGame(gameName: string) {
-    let params = {
-      TableName: table,
-      Key: {
-        gameName: gameName
-      }
-    };
-
-    try {
-      let response = await docClient.get(params).promise();
-      if (response.Item) {
-        let game = new Game(response.Item.gameName, response.Item.yearReleased, response.Item.genre, response.Item.console, response.Item.developer);
-        HttpResponse({statusCode: 200, body: JSON.stringify(game)});
-      } else {
-        throw Error;
-     }
-    } catch (err) {
-      throw HttpResponse({statusCode: 404, body: `Unable to find game.`});
     }
+}  
+
+export async function GetGameHttpResponse(game: Game) {
+  let response = await GetGame(game);
+  if (response.code) {
+    return ParseDynamoError(response.code);
+  } else if (response.Item) {
+    return HttpResponse({statusCode: 200, body: JSON.stringify(response)});
+  } else {
+    return HttpResponse({statusCode: 404, body: "Unable to get game."});
   }
-  
-  async function listGames() {
-    let params = {
-      TableName: table,
-      Select: "ALL_ATTRIBUTES"
-    };
-    
-    try {
-      let response = await docClient.scan(params).promise();
-      let gameList : Game[] = [];
-      
-      if (response.Items) {
-        response.Items.forEach((game: Game) => {
-          gameList.push(new Game(game.gameName, game.yearReleased, game.genre, game.console, game.developer));
-        })
-        HttpResponse({statusCode: 200, body: JSON.stringify(gameList)})
-      } else {
-        throw Error;
-     }
-    } catch (err) {
-      throw HttpResponse({statusCode: 404, body: `Game list could not be found.`});
-    }
+}
+
+export async function ListGamesHttpResponse() {
+  let response = await ListGames();
+  if (response.code) {
+    return ParseDynamoError(response.code);
+  } else if (response.Items) {
+    return HttpResponse({statusCode: 200, body: JSON.stringify(response)});
+  } else {
+    return HttpResponse({statusCode: 404, body: "Unable to get list of games."})
   }
+}
 
-  async function createGame(game: Game) {
-    let params = {
-      TableName: table,
-      Item: {
-        gameName: game.gameName,
-        genre: game.genre,
-        yearReleased: game.yearReleased,
-        developer: game.developer,
-        console: game.console
-      },
-      ConditionExpression: 'attribute_not_exists(gameName)'
-    }
+export async function CreateGameHttpResponse(game: Game) {
+  let response = await game.CreateGame();
+  if (response.code) {
+    return ParseDynamoError(response.code);
+  } else if (response.Item) {
+    return HttpResponse({statusCode: 201, body: JSON.stringify(response)});
+  } else {
+    return HttpResponse({statusCode: 404, body: "Unable to create game."});
```

Comment:
404 means 'Not Found' which I don't think is appropriate here. What would be the case where this block is executed? Should it just be 500?